### ISSUE TITLE
BART survey column names are messed up

### ISSUE THREAD

**Issue by jhelsel11** [MetropolitanTransportationCommission/onboard-surveys#26]
The BART survey transfer column names are a mess. They use 3 different schemes for connecting words by "_", use "trnsfr", "trnsf", and "trans" to all mean "transfer", and there may be other errors I haven't caught. For now I'm renaming column names manually in my work so that I can use regular expressions appropriately, but I think it's worth adding a column to the BART data dictionary that stores these fixes in an authoritative way.

I've opened this issue to ask @shimonisrael the best way to create that crosswalk and have it stored appropriately with the BART data and to make sure @DavidOry knows that I'm changing column names so that we can propagate the corrections downstream. Thoughts?

**Comment by shimonisrael** [comment]
@jhelsel11 - You mean transfer columns within the BART survey database (not to/from BART for other operators)? Like the first, second, third each have the name slightly different? Not certain I understand.

**Comment by jhelsel11** [comment]
The BART survey records a main trip on BART by keeping the enter BART and exit BART station. It also keeps transfer behavior for up to three transfers before and up to three transfers after the passenger is on BART. The column names of these transfers are not consistent.

The access columns are: accesstrnsfr_list1, accesstrnsfr_list2, accesstrsnfr_list3, access_**trnsf**_list1_imputed, accesstrnsf**r_l**ist2_imputed, accesstrnsf**r_l**ist3_imputed. The most egregious egress columns is egress**transr**_list3_imputed. I include the imputed column because I used it to update the primary column where there was missing information and the underscores created confusion in early passes of the code.

It wouldn't be mandatory to rename them, but I think it's worth doing for two reasons. 1) It is much easier to write flexible code if the column names vary in a predictable manner since for loops or other repetitions don't require as much customization. 2) The BART survey has 252 variables (and all of the other surveys have more than 150) so I relied upon regular expressions to help identify the variables I should isolate. I messed that process up the first time I went through the BART survey because the typos are small and unpredictable. I think future users would appreciate those typos being corrected so that they don't have to do the same trial and error that's already been solved.

**Comment by shimonisrael** [comment]
I see. Perhaps we need to discuss this on the phone. I guess I thought the standard dictionary solved that issue. I may not be understanding the work flow of the script as currently constituted. But, for BART, aren't those column names changed to the standard column names:

first_route_before_survey_board
second_route_before_survey_board
third_route_before_survey_board
first_route_after_survey_alight
second_route_after_survey_alight
third_route_after_survey_alight

**Comment by jhelsel11** [comment]
I think I've worked around the problem. I think the problem can be summarized as follows.
Goal: Replacing the generic rail trips (e.g. 'BART', 'Caltrain (Unspecified)' with a station to station pairing.
Problem: The lat/long information is not preserved in the standard dictionary, and this data is required in order to find the closest station. Therefore, the code created in standard_route_name_db.R on the canonical_names branch is not immediately useable by the 'Build Standard Database.Rmd'.
Work around: I am in the process of adding the nearest neighbor code and functions I wrote as a step within the read_operator() function. I am doing this within an if statement because some of the surveys did not store lat/long data for transfer trips (e.g., BART and Caltrain). 
Future direction: The code appears to work in that it appropriately renames the rail routes for AC Transit and SF Muni (which was what I'd accomplished in a separate script) and does not fail when evaluating the other surveys. I can't think of a better way to do this, but I can think of two immediate drawbacks to my approach. a) It requires the user to maintain a list of the combination of transfer route, boarding lat/long, and alighting lat/long (get_rail_names_inputs.csv). This effort may prove onerous. b) The if statement is not aesthetically pretty since the point of the read_operator() function is to put all of this work in the standard dictionary csv. I'm open to changing this methodology.
